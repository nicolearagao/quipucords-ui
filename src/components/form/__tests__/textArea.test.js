import React from 'react';
import { TextArea, TextAreResizeOrientation } from '../textArea';

describe('TextArea Component', () => {
  it('should render a basic component', () => {
    const props = {};

    const component = renderComponent(<TextArea {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should export support constants, types', () => {
    expect({ TextAreResizeOrientation }).toMatchSnapshot('support');
  });

  it('should handle readOnly, disabled', async () => {
    const props = {
      isReadOnly: true
    };

    const component = renderComponent(<TextArea {...props} />);
    expect(component).toMatchSnapshot('readOnly');

    const componentDisabled = component.setProps({
      isReadOnly: false,
      isDisabled: true
    });

    expect(componentDisabled).toMatchSnapshot('disabled');

    const componentActive = component.setProps({
      isReadOnly: false,
      isDisabled: false
    });

    expect(componentActive).toMatchSnapshot('active');
  });

  it('should return an emulated onChange event', () => {
    const props = {
      onChange: jest.fn(),
      value: 'lorem ipsum'
    };

    const component = renderComponent(<TextArea {...props} />);
    const input = component.find('textarea');
    const mockEvent = { target: { value: 'dolor sit' } };
    component.fireEvent.change(input, mockEvent);

    expect(props.onChange.mock.calls).toMatchSnapshot('emulated event, change');
  });

  it('should return an emulated onClear event on escape', () => {
    const props = {
      onClear: jest.fn(),
      value: 'lorem ipsum'
    };

    const component = renderComponent(<TextArea {...props} />);
    const input = component.find('textarea');
    const mockEvent = { keyCode: 27, which: 27, target: { value: '' }, key: 'Escape' };
    component.fireEvent.keyUp(input, mockEvent);

    expect(props.onClear).toHaveBeenCalledTimes(1);
    expect(props.onClear.mock.calls).toMatchSnapshot('emulated event, esc');
  });
});
