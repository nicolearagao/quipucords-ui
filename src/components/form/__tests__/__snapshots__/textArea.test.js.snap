// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component should export support constants, types: support 1`] = `
{
  "TextAreResizeOrientation": {
    "both": "both",
    "default": undefined,
    "horizontal": "horizontal",
    "none": "none",
    "vertical": "vertical",
  },
}
`;

exports[`TextArea Component should handle readOnly, disabled: active 1`] = `
<textarea
  props="{
  "isReadOnly": false,
  "className": "",
  "id": null,
  "isDisabled": false,
  "name": null,
  "value": "",
  "children": []
}"
>
  <textarea
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__textarea  "
    id="generatedid-"
    name="generatedid-"
  />
</textarea>
`;

exports[`TextArea Component should handle readOnly, disabled: disabled 1`] = `
<textarea
  props="{
  "isReadOnly": false,
  "className": "",
  "id": null,
  "isDisabled": true,
  "name": null,
  "value": "",
  "children": []
}"
>
  <textarea
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__textarea  "
    disabled=""
    id="generatedid-"
    name="generatedid-"
  />
</textarea>
`;

exports[`TextArea Component should handle readOnly, disabled: readOnly 1`] = `
<textarea
  props="{
  "isReadOnly": true,
  "className": "",
  "id": null,
  "isDisabled": false,
  "name": null,
  "value": "",
  "children": []
}"
>
  <textarea
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__textarea  "
    id="generatedid-"
    name="generatedid-"
    readonly=""
  />
</textarea>
`;

exports[`TextArea Component should render a basic component: basic 1`] = `
<textarea
  props="{
  "className": "",
  "id": null,
  "isDisabled": false,
  "isReadOnly": false,
  "name": null,
  "value": "",
  "children": []
}"
>
  <textarea
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__textarea  "
    id="generatedid-"
    name="generatedid-"
  />
</textarea>
`;

exports[`TextArea Component should return an emulated onChange event: emulated event, change 1`] = `
[
  [
    {
      "checked": undefined,
      "currentTarget": <textarea
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__textarea  "
        id="generatedid-"
        name="generatedid-"
      >
        dolor sit
      </textarea>,
      "id": "generatedid-",
      "keyCode": undefined,
      "name": "generatedid-",
      "persist": [Function],
      "target": <textarea
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__textarea  "
        id="generatedid-"
        name="generatedid-"
      >
        dolor sit
      </textarea>,
      "value": "dolor sit",
    },
  ],
]
`;

exports[`TextArea Component should return an emulated onClear event on escape: emulated event, esc 1`] = `
[
  [
    {
      "checked": undefined,
      "currentTarget": <textarea
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__textarea  "
        id="generatedid-"
        name="generatedid-"
      />,
      "id": "generatedid-",
      "keyCode": 27,
      "name": "generatedid-",
      "persist": [Function],
      "target": <textarea
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__textarea  "
        id="generatedid-"
        name="generatedid-"
      />,
      "value": "",
    },
  ],
]
`;
